### PR TITLE
Improve Stepper controlsBuilder docs

### LINKED ISSUE #57419: Stepper UX - controls per step, eg last step should be "Save" button
It is unclear to me from the docs, how one could have the last step with a save button?
https://api.flutter.dev/flutter/material/Stepper/controlsBuilder.html

### PR DESCRIPTION
**Pull request by JuYeong0413**
## Description

Improve Stepper `controlsBuilder` docs about how to change the button text according to the step.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/57419

## Tests

None, only docs were updated.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. 

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/stepper.dart [hunk 1]
```diff
@@ -195,6 +195,8 @@ class Stepper extends StatefulWidget {
   ///
   /// This callback which takes in a context and two functions: [onStepContinue]
   /// and [onStepCancel]. These can be used to control the stepper.
+  /// For example, keeping track of the [currentStep] within the callback can
+  /// change the text of the button, like 'save' at the last step.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/stepper.dart:199]: What do you mean by 'save' here?

Comment (JuYeong0413) [reply packages/flutter/lib/src/material/stepper.dart:199]: Ah, it's supposing that users are filling some forms in `Stepper`.
```suggestion
  /// change the text of the continue or cancel button as per which step are users at.
```
How about updating like this? Would it be better?

Comment (Piinks) [reply packages/flutter/lib/src/material/stepper.dart:199]: That sounds good to me! Thank you!

**Comment by JuYeong0413** [comment]
I've applied your review and updated this branch with latest master. Thank you! :)

File: packages/flutter/lib/src/material/stepper.dart [hunk 2]
```diff
@@ -195,6 +195,8 @@ class Stepper extends StatefulWidget {
   ///
   /// This callback which takes in a context and two functions: [onStepContinue]
   /// and [onStepCancel]. These can be used to control the stepper.
+  /// For example, keeping track of the [currentStep] within the callback can
+  /// change the text of the continue or cancel button as per which step are users at.
```
Comment (PixelToast) [inline packages/flutter/lib/src/material/stepper.dart:199]:
```suggestion
  /// change the text of the continue or cancel button depending on which step users are at.
```

Comment (JuYeong0413) [reply packages/flutter/lib/src/material/stepper.dart]: Applied. Thanks for your review! :)

**Comment by PixelToast** [comment]
LGTM

**Review by pingbird — APPROVED** [review]

**Pull request merged**